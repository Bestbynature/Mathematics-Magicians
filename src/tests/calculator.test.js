import { render } from '@testing-library/react';
import Calculator from '../components/calculator';

describe('Calculator', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render the component and update the display on button click', () => {
    const { getByTestId } = render(<Calculator />);
    const displayElement = getByTestId('click').textContent;
    expect(displayElement).toBe(' ');
  });
});
