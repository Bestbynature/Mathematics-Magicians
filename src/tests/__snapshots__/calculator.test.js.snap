// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render correctly 1`] = `
<DocumentFragment>
  <div
    class="calc-wrapper"
  >
    <h2
      data-testid="calc-heading"
    >
      Let's do some math!
    </h2>
    <div
      class="calculator"
    >
      <div
        class="display"
        data-testid="click"
      >
         
      </div>
      <div
        class="buttons-cont"
      >
        <button
          type="button"
        >
          AC
        </button>
        <button
          type="button"
        >
          +/-
        </button>
        <button
          type="button"
        >
          %
        </button>
        <button
          class="operations"
          type="button"
        >
          รท
        </button>
        <button
          type="button"
        >
          7
        </button>
        <button
          type="button"
        >
          8
        </button>
        <button
          type="button"
        >
          9
        </button>
        <button
          class="operations"
          type="button"
        >
          ร
        </button>
        <button
          type="button"
        >
          4
        </button>
        <button
          type="button"
        >
          5
        </button>
        <button
          type="button"
        >
          6
        </button>
        <button
          class="operations"
          type="button"
        >
          -
        </button>
        <button
          type="button"
        >
          1
        </button>
        <button
          type="button"
        >
          2
        </button>
        <button
          type="button"
        >
          3
        </button>
        <button
          class="operations"
          type="button"
        >
          +
        </button>
        <button
          class="rowspan"
          type="button"
        >
          0
        </button>
        <button
          type="button"
        >
          .
        </button>
        <button
          class="operations"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
